core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -851,8 +852,16 @@ private[spark] class MapOutputTrackerWorker(conf: SparkConf) extends MapOutputTr
         var fetchedStatuses = mapStatuses.get(shuffleId).orNull
         if (fetchedStatuses == null) {
           logInfo("Doing the fetch; tracker endpoint = " + trackerEndpoint)
-          val fetchedBytes = askTracker[Array[Byte]](GetMapOutputStatuses(shuffleId))
-          fetchedStatuses = MapOutputTracker.deserializeMapStatuses(fetchedBytes, conf)
+          try {
+            val fetchedBytes = askTracker[Array[Byte]](GetMapOutputStatuses(shuffleId))
+            fetchedStatuses = MapOutputTracker.deserializeMapStatuses(fetchedBytes, conf)
+          } catch {
+            case e: IOException if
+            Throwables.getCausalChain(e).asScala.exists(_.isInstanceOf[BlockNotFoundException]) =>
+              mapStatuses.clear()
```

Comment:
Is it OK to clear out all the map status? Shouldn't we only drop the data of the current shuffle id?

Reply:
@cloud-fan Good question! yes, I think we should just drop the data of the current shuffle id, but  seems that we currently bind an global epoch to the `MapOutputTracker`, if one stage `FetchFailed`, then the epoch will be updated, so that it will clear all the map statuses cache in the executor side.
Should we change this behavior? if so may be we can put another PR for that.

Reply:
I think there's a potential assuming that *shuffle data aways randomly and evenly placed on nodes*. That means, any shuffle fetch failure can imply the potential fetch failure for other shuffles in future. So, currently, we aways clear `mapStatuses` when fetch failure happens.

Reply:
But here is the broadcast being invalid issue. I don't think it usually happens for a lot of shuffles at the same time.